src/legacy/core_plugins/expressions/public/np_ready/public/loader.test.ts

```diff
@@ -86,6 +86,14 @@ describe('ExpressionDataHandler', () => {
     expect(response).toEqual({ type: 'render', as: 'test' });
   });
 
+  it('emits on loading$ when starting to load', async () => {
+    const expressionLoader = new ExpressionLoader(element, expressionString, {});
+    const loadingPromise = expressionLoader.loading$.pipe(first()).toPromise();
+    expressionLoader.update('test');
```

Comment:
Based on the code you've written, I don't think this test is actually testing the `update` function. It looks to me like `expressionLoader.loading$` is actually called twice, once in the constructor, and again during the update function. Left some other comments to this effect but this test doesn't accurately capture the lifecycle

Reply:
i updated the test so it checks if loading$ emits when updating as well as when calling update with no parameters (which will rerender, but will not refetch data)